src/components/card/Card.jsx

```diff
@@ -109,7 +77,36 @@ const Card = ({
     setDimensions(newStyle)
   }, [height])
 
-  const displayTwitterFooter = isTwitter && !mobileWebTwitter
+  const setCardSize = useCallback(() => {
+    // Specialcase check for mobile twitter
+    // If it's a square aspect ratio and
+    // below a certain width, we should render
+    // the card square fullscreen.
+    if (mobileWebTwitter) {
+      setCardStyle({
+        height: `${window.document.documentElement.clientHeight}px`,
+        width: `${window.document.documentElement.clientWidth}px`
+      })
+      return
+    }
+
+    const aspectRatio = isTwitter ? ASPECT_RATIOS.twitter : ASPECT_RATIOS.standard
+    const viewportAspectRatio = (window.document.body.clientWidth / window.document.body.clientHeight)
+
+    if (aspectRatio < viewportAspectRatio) {
+      // In this case, we have 'extra' width so height is the constraining factor
+      setCardStyle({
+        height: `${cardRef.current?.parentElement.clientHeight}px`,
```

Comment:
Just want to confirm this still works fine if cardRef.current is null briefly before the first paint?